src/client/windshaft.js

```diff
@@ -244,26 +244,28 @@ export default class Windshaft {
             threshold: 1
         };
 
-        MNS.columns
-            .forEach(name => {
-                if (name !== 'cartodb_id') {
-                    if (schema.column.isAggregated(name)) {
-                        aggregation.columns[name] = {
-                            aggregate_function: schema.column.getAggFN(name),
-                            aggregated_column: schema.column.getBase(name)
-                        };
-                    } else {
-                        aggregation.dimensions[name] = name;
-                    }
+        Object.keys(MNS)
+            .forEach(propertyName => {
+                if (propertyName !== 'cartodb_id') {
+                    const propertyUsages = MNS[propertyName];
+                    propertyUsages.forEach(usage => {
+                        if (usage.type === 'aggregated') {
+                            aggregation.columns[schema.column.aggColumn(propertyName, usage.op)] = {
+                                aggregate_function: usage.op,
+                                aggregated_column: propertyName
+                            };
+                        } else {
+                            aggregation.dimensions[propertyName] = propertyName;
+                        }
+                    });
                 }
             });
 
         return aggregation;
     }
 
     _buildSelectClause (MNS) {
-        const columns = MNS.columns.map(name => schema.column.getBase(name))
-            .concat(['the_geom_webmercator', 'cartodb_id']);
+        const columns = Object.keys(MNS).concat(['the_geom_webmercator', 'cartodb_id']);
```

Comment:
Is the following uniqueness filter needed now?

Reply:
I think so, the `cartodb_id` could be referenced within the viz and it would be appended now too